### ISSUE TITLE
[enhancement] User Interface improvement for Bolus Correction in calculator

### ISSUE THREAD

**Issue by Philoul** [MilosKozak/AndroidAPS#2462]
Today, there are 3 different ways to adjust the amount of insulin injected after a calculation with the bolus wizard:
- A "Corr" field in the calculator (value in Unit of Insulin which can be positive or negative)
- A “Wizard percentage” parameter in the watch (if activated) (value in % which can be greater or less than 100)
- A "Deliver this part of bolus wizard result [%]" parameter in the preferences (less than or equal to 100)

The interaction between all these parameters is not very simple:
- From the calculator: 
            ((Calculated amount of insulin) + Corr) * (Percentage of Bolus delivery)
- From the watch if the percentage of correction parameter is not activated:
            (Calculated amount of insulin)
- From the watch if the percentage of correction parameter is activated:
            (Calculated amount of insulin) * (Percentage of Correction)

Although from the watch the calculation is carried out by the bolus wizard of AAPS, the "Percentage of Bolus delivery" parameter entered in preferences is not currently taken into account, which can be confusing… (cf. https://github.com/MilosKozak/AndroidAPS/issues/2221)
Furthermore if punctually we wish to administer 100% of the calculated bolus or more (if “percentage of bolus delivery” value is less than 100% in the preferences), it is more complicated to adjust the value (entered in Unit of Insulin) in the Corr field...

I propose to simplify User Interface when we want to make the corrections in the wizard:
In the calculator, I propose to make the correction in percentage (instead of using the Unit of Insulin), and to use as default value for this "Corr" field the percentage defined in the preferences ("Deliver this part of bolus wizard result" value). That way default value of Corr field is allways less or equal to 100%.
=> The insulin calculation is simplier: 
            (Calculated amount of insulin) * (percentage of correction)
=> If I wish to adjust the correction from time to time from calculator, I can modify Corr value (with a value less than or greater than 100%). I think adjustment is easier and the result is directly shown in Insulin units in global amount of insulin calculation.

If wizard is launched from the watch, the calculation would be exactly the same:
- If the percentage of correction parameter is not activated on the watch: 
            (Calculated amount of insulin) * (Deliver this part of bolus wizard result)
- If the parameter percentage of correction is active: 
            (Calculated amount of insulin) * (percentage of correction)

**Comment by guydavies** [comment]
I would definitely not like to see the ability to simply modify the bolus by adding an absolute quantity removed. I use this method frequently (daily) and don't want to be forced to calculate the percentage required to raise/drop the quantity delivered to my preferred amount. That would be a regression in my view.

**Comment by Philoul** [comment]
I also often use correction, everytime I change a value in calculator, I can see immediatly the amount of insulin in Result.
=> So you don't need to calculate the percentage of correction, i.e. if you see in "Result" 6U and you want to have 7U, it's easy and quick to increase percentage with a few press on + button to reach this value.
=> And if you want to enter both carbs and insulin with preferred amount, there is also Treatment button to directly enter these values (quicker than in calculator...)

=> I set in preferences Deliver this part of bolus wizard result [%] at 80% to be safe (and I let SMB adjust). That way if I under-estimate the amount of carb or if I do a little more activities I have less risk of hypo... But when I want to have exact amount of my profil with current IOB, COB adjustments, then it's complicated to calculate because lots of correction are done by wizard and you don't see the results of calculation without global percentage...

Do you use a Wear OS watch to control AAPS ?
I ask you this question because in watch, correction is done with percentage, and using both logic (in percentage from watch and in Insulin Units in calculator), I think that percentage is easier to use...

**Comment by guydavies** [comment]
I do use a wear watch but only for TBR. I don't bolus from the watch.

I simply insert the variation directly into the calculator. The calculator
provides me with the base value and I insert the positive or negative
difference.

What you consider "easier" may not be considered so by everyone. YDMV and
it's important that we don't remove options simply because one person feels
they're unnecessary.

**Comment by MilosKozak** [comment]
everything has a reason. we have these 2 splitted controls because they are intended to do different thing.
percentace -> from SMB. do not do full bolus, let the alg do it's job
corr -> cover current situation like sensitivity change, activity, kind of food etc

That's the reason why percentage is in prefereces and not easy accessible for change. And correction from my point of view is easier to do in absolute values

**Comment by Philoul** [comment]
Thank you Milos for this explanation.

@guydavies It was just a suggestion for improvement because I thought it was better, but I can understand that it is not suitable for everyone and if there is indeed a good reason and it's easier to a majority of people to make the correction in absolute value, this is not a problem for me.

However, I do not understand why the preference setting "percentage of bolus delivery" is not taken into account when calculation is sent from watch,  and why correction is in percentage and not in U if it's easier ?
=> If it's for SMB it should be, no ?
=> And then may be it's in watch that wizard correction needs to be switched from % to U to be consistent...

No problem for me to close this proposal if you think it's not a good idea and nothing should be changed either in the watch or in the calculator ;-)

**Comment by guydavies** [comment]
I agree that a preset percentage should be applied (with a notification in
case you forgot it was applied)

**Comment by wootmasterslick** [comment]
I second any suggestions to simplify the bolus calculator. In its current state it can be very complicated/confusing for novice users. With this idea we make a step to make it more understandable. I also agree that we should not remove the option to correct the bolus in units. A way to still do this and implement your idea would to adjust the corr in either % or Insulin punits and then directly beside it display the opposite value. So for example the calculated amount is 10 U and in the corr field I enter -2 U then beside it would display 80%
